nixos/modules/services/games/minecraft-server.nix

```diff
@@ -4,6 +4,37 @@ with lib;
 
 let
   cfg = config.services.minecraft-server;
+
+  eulaFile = pkgs.writeText "eula.txt" ''
+    # eula.txt managed by NixOS Configuration
+    eula=${lib.boolToString cfg.eula}
+  '';
+
+  whitelistFile = pkgs.writeText "whitelist.json"
+    (builtins.toJSON
+      (lib.mapAttrsToList (n: v: { name = n; uuid = v; }) cfg.whitelist));
+
+  # See https://minecraft.gamepedia.com/Server.properties
+  # for description of each property and allowed values
+  # these are the minimum required for opening correct ports
+  defaultServerProperties = {
+    enable-query = false;
+    enable-rcon = false;
+    "query.port" = 25565;
+    "rcon.port" = 25575;
+    server-port = 25565;
+  };
```

Comment:
As mentioned previously, you should put these in the `config` section. Something like this to be exact:

```nix
services.minecraft-server.serverProperties = {
  enable-query = mkDefault false;
  enable-rcon = mkDefault false;
  "query.port" = mkDefault 25565;
  "rcon.port" = mkDefault 25575;
  server-port = mkDefault 25565;
};
```

Then you can remove the manual merging below and use `cfg.serverProperties` directly, which will have everything merged already.

Reply:
Fixed.